datastream/dataset.py

```diff
@@ -66,17 +94,36 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def __add__(self, other):
+    def __add__(self, other: Dataset) -> Dataset:
         return Dataset.concat([self, other])
 
-    def map(self, function):
+    def map(self, function: Callable) -> Dataset:
+        '''Append a function to the dataset pipeline.'''
```

Comment:
`append` is an implementation detail

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.map.html

Reply:
Maybe. I chose 'append' to show were the function is added (could be in front)

Got any suggestions? I don't think the pd.Series documentation is readable

Reply:
The only sensible order is to call them in the order they where "mapped"

Maps the datasets <elements> through `function` to generate a new dataset
alt.
Generates a Dataset where the elements are `function(element)`